test/unit/theory/theory_quantifiers_bv_instantiator_white.h

```diff
@@ -0,0 +1,416 @@
+/*********************                                                        */
+/*! \file theory_bv_bvgauss_white.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Mathias Preiner
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2017 by the authors listed in the file AUTHORS
+ ** in the top-level source directory) and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Unit tests for BvInstantiator.
+ **
+ ** Unit tests for BvInstantiator.
+ **/
+
+#include "expr/node.h"
+#include "expr/node_manager.h"
+#include "smt/smt_engine.h"
+#include "smt/smt_engine_scope.h"
+#include "theory/rewriter.h"
+#include "theory/bv/theory_bv_utils.h"
+#include "util/bitvector.h"
+
+#include "theory/quantifiers/ceg_t_instantiator.cpp"
+
+#include <cxxtest/TestSuite.h>
+#include <iostream>
+#include <vector>
+
+using namespace CVC4;
+using namespace CVC4::theory;
+using namespace CVC4::theory::bv;
+using namespace CVC4::theory::bv::utils;
+using namespace CVC4::theory::quantifiers;
+using namespace CVC4::smt;
+
+class BvInstantiatorWhite : public CxxTest::TestSuite
+{
+ public:
+  void setUp();
+  void tearDown();
+
+  void testGetPvCoeff();
+  void testNormalizePvMult();
+  void testNormalizePvPlus();
+  void testNormalizePvEqual();
+
+ private:
+  ExprManager *d_em;
+  NodeManager *d_nm;
+  SmtEngine *d_smt;
+  SmtScope *d_scope;
+
+  Node mkNeg(TNode n);
+  Node mkMult(TNode a, TNode b);
+  Node mkMult(const std::vector<Node>& children);
+  Node mkPlus(TNode a, TNode b);
+  Node mkPlus(const std::vector<Node>& children);
+};
+
+void BvInstantiatorWhite::setUp()
+{
+  d_em = new ExprManager();
+  d_nm = NodeManager::fromExprManager(d_em);
+  d_smt = new SmtEngine(d_em);
+  d_scope = new SmtScope(d_smt);
+}
+
+void BvInstantiatorWhite::tearDown()
+{
+  delete d_scope;
+  delete d_smt;
+  delete d_em;
+}
+
+Node BvInstantiatorWhite::mkNeg(TNode n)
+{
+  return d_nm->mkNode(kind::BITVECTOR_NEG, n);
+}
+
+Node BvInstantiatorWhite::mkMult(TNode a, TNode b)
+{
+  return d_nm->mkNode(kind::BITVECTOR_MULT, a, b);
+}
+
+Node BvInstantiatorWhite::mkMult(const std::vector<Node>& children)
+{
+  return d_nm->mkNode(kind::BITVECTOR_MULT, children);
+}
+
+Node BvInstantiatorWhite::mkPlus(TNode a, TNode b)
+{
+  return d_nm->mkNode(kind::BITVECTOR_PLUS, a, b);
+}
+
+Node BvInstantiatorWhite::mkPlus(const std::vector<Node>& children)
+{
+  return d_nm->mkNode(kind::BITVECTOR_PLUS, children);
+}
+
+/**
+ * getPvCoeff(x, n) should return
+ *
+ *  1           if n == x
+ * -1           if n == -x
+ *  a           if n == x * a
+ * -a           if n == x * -a
+ * Node::null() otherwise.
+ *
+ * Note that x * a and x * -a have to be normalized.
+ */
+void BvInstantiatorWhite::testGetPvCoeff()
+{
+  Node x = mkVar(32);
+  Node a = mkVar(32);
+  Node one = mkOne(32);
```

Comment:
It seems that `x`, `a`, `b`, `c`, `d`, `zero` and `one` are always initialized as the same expressions. Why not have members `d_x`, `d_a`, `d_b`, `d_c`, `d_d`, `d_zero` and `d_one` which you can then initialize in `setUp` and destroy in `tearDown`? Just be careful to set them to `Node::null()` before you delete `d_em`.

Reply:
I'll keep it as is since I prefer in these tests to have shorter variable names (without the d_* prefix*).